src/common/ad_mem.c

```diff
@@ -318,6 +308,43 @@ blob_set_opened(struct ad_blob *blob)
 	}
 }
 
+static void
+blob_close(struct ad_blob *blob)
+{
+	struct ad_arena *arena;
+	struct ad_group *group;
+
+	D_ASSERT(blob->bb_opened > 0);
+	D_DEBUG(DB_TRACE, "Close blob, openers=%d\n", blob->bb_opened);
+	blob->bb_opened--;
+	if (blob->bb_opened > 0)
+		return;
+
+	D_DEBUG(DB_TRACE, "Evict unpublished groups and arenas\n");
+	while ((group = d_list_pop_entry(&blob->bb_gps_rsv, struct ad_group, gp_link))) {
+		D_ASSERT(group->gp_unpub);
+		group_decref(group);
+	}
+
+	while ((arena = d_list_pop_entry(&blob->bb_ars_rsv, struct ad_arena, ar_link))) {
+		D_ASSERT(arena->ar_unpub);
+		arena_decref(arena);
+	}
+
+	D_DEBUG(DB_TRACE, "Unbind groups and arenas in LRU\n");
+	d_list_for_each_entry(group, &blob->bb_gps_lru, gp_link)
+		group_unbind(group, false);
```

Comment:
In group_unbind(similar as following arena_unbind) we will modify gp_df->gd_back_ptr, there wont change to write it back, so we are ok to ignore this update right, a bit confusing though.

Reply:
yes, this is how back_ptr works, we don't need it to be written back to storage. 
We use the durable arena & group, which is addressed by offset, to index in-memory data structure. So we don't have to create another huge tree or hash table to index them.
"Incarnation" is the way that we can detect stale pointers.